### ISSUE TITLE
Query regarding debiased model saving in Adversarial Debiasing

### ISSUE THREAD

**Issue by SBGitHub21** [Trusted-AI/AIF360#286]
Hello, is there a way to save and reuse the debiased model created in Adversarial Debiasing? 
This model:
<img width="479" alt="model" src="https://user-images.githubusercontent.com/40518134/148887330-6659de20-5f9c-4899-bb65-44c736f75a01.png">
I have tried pickle , joblib and keras method, none of them seem to work.
<img width="542" alt="tried" src="https://user-images.githubusercontent.com/40518134/148887449-ef58577a-4eec-41dd-b5fa-be2877ac59d3.png">

**Comment by giandos200** [comment]
try to use the Sklearn version instead of the TF version and you will have no problem with pickle dump after the fit. if you want to use the TF version try to save it as a TF model.

**Comment by SBGitHub21** [comment]
I am trying to use the sklearn version but it gets stuck at :
<img width="784" alt="err" src="https://user-images.githubusercontent.com/40518134/152784708-3f886a29-6250-44fd-8255-0bc0e2e71041.png">
I have tried sending the protected attribute as a string, a list as well as leaving it None. Nothing seems to work. Where am I going wrong? I have split the data using sklearn train test split as well as tried converting the data into BinaryDataset type. 

As far as tensorflow model saving goes, those methods are not working either.
<img width="821" alt="err2" src="https://user-images.githubusercontent.com/40518134/152785108-e0b9df8e-d816-4032-a1d7-ef1be8ea2da8.png">

**Comment by evilantal** [comment]
I tried this using the sklearn version. But when I try to pickle.dump() using cloudpickle, I get an error saying "cannot pickle 'Graph' object".
I'm not even sure why a fitted AdversarialDebiasing model would be a 'Graph' object.

cloudpickle version = 1.5.0
aif360 version = 0.5.0

In the example below X is a DataFrame that includes 'protfeat' set as index and y is a numpy array of 1/0 labels.

```
from aif360.sklearn.inprocessing import AdversarialDebiasing
import tensorflow.compat.v1 as tf
from aif360.datasets import StandardDataset
import pandas as pd
from cloudpickle import pickle

tf.disable_eager_execution()
sess = tf.Session()

model = AdversarialDebiasing(
    prot_attr='protfeat',
    scope_name='debiased_model',
    num_epochs=50,
    batch_size=128,
    debias=True, 
    random_state=1337,
    classifier_num_hidden_units=200
)

model.fit(X, y)

with open(os.path.join(folder.get_path(), 'test_pickle.pkl'), 'w') as f:
     pickle.dump(model, f)
```
